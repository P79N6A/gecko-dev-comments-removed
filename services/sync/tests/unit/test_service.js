Cu.import("resource://weave/log4moz.js");
Cu.import("resource://weave/async.js");
Cu.import("resource://weave/crypto.js");

let __fakePrefs = {
  "log.logger.async" : "Debug",
  "username" : "foo",
  "serverURL" : "https://example.com/",
  "encryption" : "aes-256-cbc",
  "enabled" : true,
  "schedule" : 0
};

let __fakeDAVContents = {
  "meta/version" : "2",
  "private/privkey" : "fake private key"
};

let __fakePasswords = {
  'Mozilla Services Password': {foo: "bar"},
  'Mozilla Services Encryption Passphrase': {foo: "passphrase"}
};

let Service = loadInSandbox("resource://weave/service.js");

function TestService() {
  this.__superclassConstructor = Service.WeaveSvc;
  this.__superclassConstructor([]);
}

TestService.prototype = {
  _initLogs: function TS__initLogs() {
    this._log = Log4Moz.Service.getLogger("Service.Main");
  }
};
TestService.prototype.__proto__ = Service.WeaveSvc.prototype;

function test_login_works() {
  var fds = new FakeDAVService(__fakeDAVContents);
  var fprefs = new FakePrefService(__fakePrefs);
  var fpasses = new FakePasswordService(__fakePasswords);
  var fts = new FakeTimerService();
  var logStats = initTestLogging();
  var testService = new TestService();
  var finished = false;
  var successful = false;
  var onComplete = function(result) {
    finished = true;
    successful = result;
  };

  testService.login(onComplete);

  while (fts.processCallback()) {}

  do_check_true(finished);
  do_check_true(successful);
  do_check_eq(logStats.errorsLogged, 0);
  do_check_eq(Async.outstandingGenerators.length, 0);
}
