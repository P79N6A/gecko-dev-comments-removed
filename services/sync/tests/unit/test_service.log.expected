*** test pending
Running test: test_login_works
Service.Main	INFO	Weave Sync Service Initializing
Service.Main	DEBUG	Logging in user foo
Service.Main	INFO	Using server URL: https://example.com/user/foo
Testing	INFO	HTTP GET from meta/version, returning status 200
Service.Main	TRACE	Retrieving keypair from server
Testing	INFO	HTTP GET from private/privkey, returning status 200
Testing	INFO	HTTP GET from public/pubkey, returning status 200
Service.Main	INFO	Weave scheduler disabled
1 of 1 tests passed.
*** test finished
*** exiting
*** PASS ***
